### ISSUE TITLE
crash with "map in camera" and camera button

### ISSUE THREAD

**Issue by HolgerJeromin** [mapillary/mapillary_issues#1651]
I have a reproducable bug while having the small map active. After pushing the camera button the app crashes/restarts.

````
01-06 11:49:43.977 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onResponse
01-06 11:49:43.977 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onResponse
01-06 11:49:43.984 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onResponse
01-06 11:49:44.057 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onFailure: Canceled
01-06 11:49:44.076 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onFailure: Canceled
01-06 11:49:44.094 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onFailure: Canceled
01-06 11:49:44.117 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onFailure: Canceled
01-06 11:49:44.181 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onFailure: Canceled
01-06 11:49:44.408 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onResponse
01-06 11:49:44.578 D/com.mapbox.mapboxsdk.http.HTTPContext$HTTPRequest(27595): onResponse
01-06 11:49:49.657 D/app.mapillary.android.activity.CameraActivity(27595): Camera button pressedKeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_CAMERA, scanCode=766, metaState=0, flags=0x8, repeatCount=0, eventTime=63992060, downTime=63992060, deviceId=4, source=0x101 }
01-06 11:49:49.986 D/app.mapillary.android.activity.CameraActivity(27595): Camera button pressedKeyEvent { action=ACTION_UP, keyCode=KEYCODE_CAMERA, scanCode=766, metaState=0, flags=0x8, repeatCount=0, eventTime=63992340, downTime=63992060, deviceId=4, source=0x101 }
01-06 11:49:49.988 D/app.mapillary.android.activity.CameraActivity(27595): Free/Total disk: 14735MB/25637.0MB, percent: 57.475525
01-06 11:49:49.988 D/app.mapillary.android.activity.CameraActivity(27595): takeShot Queue/Tasks: 0/0 of executor pool size 4
01-06 11:49:49.989 D/app.mapillary.android.activity.CameraActivity(27595): Moving speed in m/s: 0.8334
01-06 11:49:49.991 D/app.mapillary.android.activity.CameraActivity(27595): requesting takePicture with focus mode infinity
01-06 11:49:49.991 D/app.mapillary.android.activity.CameraActivity(27595): requesting takePicture
01-06 11:49:50.356 D/app.mapillary.android.activity.CameraActivity(27595): shutterCallback
01-06 11:49:50.418 D/app.mapillary.android.activity.CameraActivity(27595): RawCallback.onPictureTaken
01-06 11:49:50.569 D/app.mapillary.android.activity.CameraActivity(27595): pictureCallback
--------- beginning of crash
01-06 11:49:50.598 F/libc    (27595): Fatal signal 6 (SIGABRT), code -6 in tid 27595 (app.mapillary)
01-06 11:49:50.598 I/libc    (27595): Suppressing debuggerd output because prctl(PR_GET_DUMPABLE)==0
01-06 11:49:51.493 I/Fabric  (23479): Initializing Crashlytics 2.3.3.61
01-06 11:49:51.527 D/app.mapillary.android.activity.MainActivity(23479): refreshing tokens
01-06 11:49:51.548 W/ContextImpl(23479): Failed to ensure directory: /storage/sdcard1/Android/data/app.mapillary/files
01-06 11:49:51.548 D/app.mapillary.android.service.StorageUtils(23479): Android 19+ : Context.getExternalFilesDirs(): /storage/emulated/0/Android/data/app.mapillary/files
01-06 11:49:51.548 D/app.mapillary.android.service.StorageUtils(23479): Android 19+ : Context.getExternalFilesDirs(): null
01-06 11:49:51.548 D/app.mapillary.android.service.StorageUtils(23479): bestPath: /storage/emulated/0/Android/data/app.mapillary/files/mapillary
01-06 11:49:51.548 D/app.mapillary.android.service.StorageUtils(23479): getDefaultImageStoragePath- /storage/emulated/0/Android/data/app.mapillary/files/mapillary
01-06 11:49:51.549 W/Fabric  (23479): Could not call getId on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
01-06 11:49:51.549 W/Fabric  (23479): Could not call isLimitAdTrackingEnabled on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
01-06 11:49:51.597 W/Fabric  (23479): Could not call getId on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
01-06 11:49:51.597 W/Fabric  (23479): Could not call isLimitAdTrackingEnabled on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
01-06 11:49:51.765 D/app.mapillary.android.activity.MainActivity(23479): KeyHash:rJ3uTbwQBnhr6+9El6ZuWUbdkmw=
01-06 11:49:51.780 D/app.mapillary.android.activity.MainActivity(23479): Android version: 22
01-06 11:49:51.780 D/app.mapillary.android.activity.MainActivity(23479): OS version: 3.4.0-gcd7fdcf
01-06 11:49:51.780 D/app.mapillary.android.activity.MainActivity(23479): Device FP2
01-06 11:49:51.780 D/app.mapillary.android.activity.MainActivity(23479): Model FP2
01-06 11:49:51.780 D/app.mapillary.android.activity.MainActivity(23479): Product FP2
01-06 11:49:51.781 D/app.mapillary.android.service.StorageUtils(23479): getDefaultImageStoragePath- /storage/emulated/0/Android/data/app.mapillary/files/mapillary
01-06 11:49:51.845 D/app.mapillary.android.service.StorageUtils(23479): getDefaultImageStoragePath- /storage/emulated/0/Android/data/app.mapillary/files/mapillary
01-06 11:49:51.848 D/app.mapillary.android.activity.MainActivity(23479): Photos left to upload: 419 in 18 sequences
01-06 11:49:51.861 D/app.mapillary.android.activity.MainActivity(23479): Mapillary versionName: 1.69
01-06 11:49:51.861 D/app.mapillary.android.activity.MainActivity(23479): Mapillary versionCode: 169
01-06 11:49:51.869 D/app.mapillary.android.activity.MainActivity(23479): Avatar image not found
01-06 11:49:51.908 W/Fabric  (23479): Could not call getId on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
01-06 11:49:51.908 W/Fabric  (23479): Could not call isLimitAdTrackingEnabled on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
01-06 11:49:52.026 D/app.mapillary.android.service.StorageUtils(23479): getDefaultImageStoragePath- /storage/emulated/0/Android/data/app.mapillary/files/mapillary
01-06 11:49:52.146 D/app.mapillary.android.service.MapillaryServerFacade(23479): getHttpPost: https://a.mapillary.com/v2/ua/login?client_id=MkJKbDA0bnZuZlcxeTJHTmFqN3g1dzo0ZmYxN2MzMTRlYzM1M2E2
01-06 11:49:52.212 D/OpenGLRenderer(23479): Use EGL_SWAP_BEHAVIOR_PRESERVED: true
01-06 11:49:52.236 D/Atlas   (23479): Validating map...
01-06 11:49:52.289 I/Adreno-EGL(23479): <qeglDrvAPI_eglInitialize:410>: EGL 1.4 QUALCOMM build:  ()
01-06 11:49:52.289 I/Adreno-EGL(23479): OpenGL ES Shader Compiler Version: E031.26.00.04
01-06 11:49:52.289 I/Adreno-EGL(23479): Build Date: 08/28/15 Fri
01-06 11:49:52.289 I/Adreno-EGL(23479): Local Branch: 
01-06 11:49:52.289 I/Adreno-EGL(23479): Remote Branch: 
01-06 11:49:52.289 I/Adreno-EGL(23479): Local Patches: 
01-06 11:49:52.289 I/Adreno-EGL(23479): Reconstruct Branch: 
01-06 11:49:52.291 I/OpenGLRenderer(23479): Initialized EGL, version 1.4
01-06 11:49:52.349 D/OpenGLRenderer(23479): Enabling debug mode 0
01-06 11:49:52.434 W/Settings(23479): Setting airplane_mode_on has moved from android.provider.Settings.System to android.provider.Settings.Global, returning read-only value.
01-06 11:49:52.655 I/Timeline(23479): Timeline: Activity_idle id: android.os.BinderProxy@9596d63 time:63995061
01-06 11:49:53.108 D/app.mapillary.android.service.MapillaryServerFacade(23479): getTokens: HTTP/1.1 200 OK
01-06 11:49:53.109 D/app.mapillary.android.service.MapillaryServerFacade(23479): getHttpGet: https://a.mapillary.com/v2/me
01-06 11:49:53.611 D/app.mapillary.android.service.MapillaryServerFacade(23479): projects:[]
01-06 11:49:53.611 D/app.mapillary.android.service.MapillaryServerFacade(23479): sign_in jwtToken:eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJtcHkiLCJzdWIiOiJKc19hblJMSC1nNlZ1MGk3US1MWm93IiwiYXVkIjoiTWtKS2JEQTBiblp1WmxjeGVUSkhUbUZxTjNnMWR6bzBabVl4TjJNek1UUmxZek0xTTJFMiIsImlhdCI6MTQ0NzQwMjkxMCwianRpIjoiMjViZGVjMDIyNWZkYzA0NGZlZWQ0NTMxZjVhNzMwODkiLCJzY28iOlsibWFwaWxsYXJ5OnVzZXIiXSwidmVyIjoxfQ.KJNdsyH0OQEHuAx4NeYxSoia5b-z5m2ooxDZh9IgkZE, upload_token: xKsbycWdMo8LHamxnSaMbg
01-06 11:49:54.366 D/app.mapillary.android.activity.FeedFragment$FetchFeedTask(23479): got: 5 kb
01-06 11:49:58.276 D/app.mapillary.android.activity.MainActivity(23479): Appending log to /storage/emulated/0/mapillary.log

````

**Comment by simonmikkelsen** [comment]
I can reproduce it too, every time. I have not had the time to make log files.

**Comment by peterneubauer** [comment]
Does the map appear or is it crashing when you press a button?

**Comment by simonmikkelsen** [comment]
The map appears and works. The crash is exactly when I press the button to take the first picture.

**Comment by simonmikkelsen** [comment]
Just reproduced it and made this log:

[mapillary_trace.txt](https://github.com/mapillary/mapillary_issues/files/81641/mapillary_trace.txt)

**Comment by HolgerJeromin** [comment]
Walking in manual image mode, taking images with the camera button. Working normal. Enabling the map works, the map is shown. pushing then the camera button the app crashes.

**Comment by HolgerJeromin** [comment]
Probably an [mapbox SDK issue](https://github.com/mapillary/mapillary_issues/issues/1639#issuecomment-170007047), too. 
I am not yet sure, but i think this crashes only with an bluetooth gps.

**Comment by simonmikkelsen** [comment]
I use the phones internal GPS and it crashes every time.

**Comment by peterneubauer** [comment]
Mmmh,
this is most probably Mapbox not behaving. Sorry, we have to wait for a
better version, ping @zugaldia
<https://github.com/orgs/mapillary/people/zugaldia>

**Comment by zugaldia** [comment]
cc: @tobrun

**Comment by tobrun** [comment]
@peterneubauer 
I want to take a look at this but can't seem to find the actual crash in loggings added to this thread. 
Could you reproduce this and add a trace?

**Comment by peterneubauer** [comment]
It seems the Mapbox lib crashes are not giving any stack traces that we can catch with app logs or crashalytics that we use to report errors. Instead, the app crashes further down and you need `adb logcat` to see it. We use v2.3.0, I'm upgrading now to 3.0.0 which might be a lot more stable, judging from the iOS experience, so just wait and we will see, it will be out in Mapillary v1.71 shortly.

**Comment by zugaldia** [comment]
@peterneubauer :+1:

**Comment by jaakkoh** [comment]
OK. Good to see that I'm not alone with the problem. I just whined about this in the blog post -- this would have obviously been a better place for that to begin with.
Look forward to getting this working! Great feature. Even better if it was also available in offline mode. .. Need to try out screen splitting and running OsmAnd on the other side of the screen with the Mapillary traces overlay -- even though that hasn't been updates at least since last summer as discussed in #1366

**Comment by peterneubauer** [comment]
Going to release the next alpha version with mapbox sdk 3.0 soon, hope that
foxes the problem!

**Comment by jaakkoh** [comment]
> just wait and we will see, it will be out in Mapillary v1.71 shortly.

@peterneubauer Sorry to report but v.1.71 gives the same undesired behavior (on my Galaxy S5).

**Comment by peterneubauer** [comment]
That's a case for @zugaldia then?

**Comment by zugaldia** [comment]
@peterneubauer considering that the map shows, but the app crashes when the camera intent is launched, this seems like a case of unhandled lifecycle methods. Could you check our [`MapFragment`](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapFragment.java) class and check if all the methods (`onCreate()`, `onStart()`, `onResume()`, `onPause()`, `onStop()`, `onDestroy()`) are implemented on your end?

/cc: @tobrun

**Comment by tobrun** [comment]
@peterneubauer, I just took a look at the related fragment in `app/mapillary/android/activity/MapboxMap.java`:

It seems that the hunch from @zugaldia  is correct. You are missing the different lifecyles method required to use `MapView` in that way. There are 2 possible solutions for this problem:

 - You implement the different lifecycle methods as seen here
 - You use the wrapper fragment implementation

For this project I would suggest going for the former. 

Could you add these methods to your `MapboxMap` class?

```java
   @Override
    public void onStart() {
        super.onStart();
        mapView.onStart();
    }

    @Override
    public void onResume() {
        super.onResume();
        mapView.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
        mapView.onPause();
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        mapView.onSaveInstanceState(outState);
        super.onSaveInstanceState(outState);
    }

    @Override
    public void onStop() {
        super.onStop();
        mapView.onStop();
    }

    @Override
    public void onLowMemory() {
        mapView.onLowMemory();
        super.onLowMemory();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mapView.onDestroy();
        mapView = null;
    }

```


/cc: @zugaldia

**Comment by peterneubauer** [comment]
Thanks a lot @tobrun! Have added, this, please try out http://android-releases.mapillary.io/app-release_171.apk @HolgerJeromin @jaakkoh and report back!

**Comment by peterneubauer** [comment]
Feel free to reopen if this reoccurs.

**Comment by peterneubauer** [comment]
v 1.72 is out in a few hours and has this fix.

**Comment by HolgerJeromin** [comment]
I have 1.75 (? :-) since yesterday and no change. 
Mode: manual foot.
GPS via Bluetooth or native. 
Normal cam view works 
Small map or big map gives crashes

**Comment by HolgerJeromin** [comment]
Same with foot auto,  riding and panorama

**Comment by HolgerJeromin** [comment]
The trace has no information, even starting the map view is not noted in the trace or debug (tested both).

**Comment by peterneubauer** [comment]
That one has the suggested changes. On that case this issue is not
resolved.

**Comment by jaakkoh** [comment]
Just adding that my experience is the same as @HolgerJeromin 's. Tested only with Galaxy S5, seems that the issue is for all devices.

Hope to see this very very nice idea mature into a functional feature. It could even eventually be developed into a map application with Mapillary "embedded". Or at least have the ability to continue using the device as _some_ sort of navigation system (for more than finding only the unmapillaried roads).

**Comment by tobrun** [comment]
I just retested with a Nexus 5 and was not able to reproduce. I'm seeing that the used dependency is `3.0.0`. Would it be possible to upgrade to `3.2.0` and preferable to `4.0.0-SNAPSHOT`?

**Comment by rsahlin** [comment]
Will update to 3.2.0 - the 4.0.0-SNAPSHOT does not seem to be available through the maven repo.

**Comment by zugaldia** [comment]
> the 4.0.0-SNAPSHOT does not seem to be available through the maven repo.

@rsahlin You need to add `maven { url "http://oss.sonatype.org/content/repositories/snapshots/" }` to your list of repositories. The full instructions are here: https://www.mapbox.com/android-sdk/#gradle

**Comment by rsahlin** [comment]
@zugaldia Thanks, will add and include in next release.
If you are beta developer you should get the 1.76 release anytime soon - otherwise you can fetch it from the mapillary site. (link at the bottom of mapillary.com)

**Comment by HolgerJeromin** [comment]
my first test looks good. 1.76 works with small and big map.

**Comment by rsahlin** [comment]
Good to hear :-)
Closing this issue, please reopen if problem persists.

**Comment by rsahlin** [comment]
Tested on Xperia Tablet Z2 and the issue is still there. Opening issue and will try with 4.0.0-SNAPSHOT.

**Comment by rsahlin** [comment]
@zugaldia; @tobrun
I have upgraded to 4.0.0-SNAPSHOT.
For our project the maven repo link - HAD to be in the project top level build.gradle.
- Maybe you could update your instructions to reflect this in order to avoid hair tearing for other devs :-)

Sadly there are build errors.
The MapView class has moved but more importantly some methods are not there anymore.
Could you please point us to 4.0.0 documentation and advice us how to transition?
Best regards
/Richard

**Comment by rsahlin** [comment]
Closing this until I get feedback on how to migrate to MapBox 4.0.0

**Comment by jaakkoh** [comment]
I am also happy to report that the camera-in-app doesn't crash anymore on my Galaxy S5. ... but:
The large map mode doesn't show a large map. It only shows the same small map in the bottom left corner of the space/area that is reserved for the large map on the screen. The photo viewer is shown properly in the top left corner. .. This is of course a new issue and should be reported as such.

**Comment by rsahlin** [comment]
Hi Jaako and thanks for reporting :-)

I have also seen problems with the large map on an Xperia Z2 Tablet, I'm guessing it will be solved when updating to MapBox 4.0.0
